### PR TITLE
fix(1905): PR builds to get cache at job level cache scope

### PR DESCRIPTION
**Pull request by parthasl**
## Context

PR builds get cache fails for job level cache scope.

## Objective

This PR will fix get cache issue at job level scope, regex pattern to identify PR and mount read-only job cache volume.

## References

https://github.com/screwdriver-cd/screwdriver/issues/1905

## License

I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -266,8 +268,20 @@ class K8sVMExecutor extends Executor {
...
+        let volumeReadOnly = false;
+
+        if (matched && matched.length === 2) {
+            const decodedToken = jwt.decode(token, { complete: true });
+
+            volumeReadOnly = true;
+            jobId = hoek.reach(decodedToken.payload,
+                'prParentJobId', { default: jobId });
+        }
+        //
```
Comment (jithine) [inline index.js]: please remove this extra comment line

File: index.js [hunk 2]
```diff
@@ -266,8 +268,20 @@ class K8sVMExecutor extends Executor {
...
+        // PRs - set pipeline, job cache volume readonly and job cache dir to parent job cache dir
+        const regex = /^PR-([0-9]+)(?::[\w-]+)?$/gi;
+        const matched = regex.exec(jobName);
+        let volumeReadOnly = false;
+
+        if (matched && matched.length === 2) {
+            const decodedToken = jwt.decode(token, { complete: true });
+
+            volumeReadOnly = true;
+            jobId = hoek.reach(decodedToken.payload,
```
Comment (jithine) [inline index.js]: what's the implication of overriding jobId itself ? Won't this cause problems for further processing of the job? For example now `launcher` might set parentJobId as jobId for PR builds.

Comment (parthasl) [reply index.js]: No, it will not, this is used only for cache dir. launcher or store-cli doesnt depend on this.

File: index.js (hunk 1 above)
Comment (parthasl) [reply index.js]: done

File: index.js (hunk 2 above)
Comment (jithine) [reply index.js]: cool, my only worry is if in future if someone misuses this variable and see unexpected results.

Comment (parthasl) [reply index.js]: agree, verified we don't pass jobId to store-cli and launcher, only the dir is passed.

**Pull request merged**